dot/sync/chain_sync.go

```diff
@@ -711,9 +711,19 @@ func handleReadyBlock(bd *types.BlockData, pendingBlocks DisjointBlockSet, ready
 	// if we're expecting headers, validate should ensure we have a header
 	if bd.Header == nil {
 		block := pendingBlocks.getBlock(bd.Hash)
+		if block == nil {
+			logger.Criticalf("block with unknown header is ready: hash=%s", bd.Hash)
+			return
+		}
+
 		bd.Header = block.header
 	}
 
+	if bd.Header != nil {
```

Comment:
I tried this branch on my local devnet and it wasn't finalising blocks.  After closer inspection, if `bd.Header == nil` and then the header is copied, wouldn't this if condition always be true?

Reply:
sorry i messed up the if statement here when i was updating it, should have been `== nil`, please pull again and retry!